mangoPURE/models/metrics.py

```diff
@@ -0,0 +1,30 @@
+from .base import TimedLoss
+import torch
+from ..data.utils import TimedAudioBatch
+
+
+class SigmoidTimedLoss(TimedLoss):
+    def __init__(self, info_type: str = "noise"):
+        """
+        :param info_type: ["noise", "speaker"] selects which metadata from batch to use
+        """
+        self.info_type = info_type
+
+    def __call__(self, batch: TimedAudioBatch) -> torch.Tensor:
+        mask = torch.zeros(batch.audio.shape)
+        for i, length in enumerate(batch.lengths):
+            mask[i, :length, :] = 1.0
+        labels = torch.zeros(batch.audio.shape)
+        labels[:, :, 0] = 1.0
+        batch_segments = None
+        if self.info_type == "noise":
+            batch_segments = batch.noises_info
+        elif self.info_type == "speaker":
+            batch_segments = batch.speakers_info
```

Comment:
Both are not supported?

Reply:
Heh, I haven't thought about that. But no, I think I have to delete this **info_type** field and just check if **noises_info** or **speakers_info*8 are present in batch, then any combination will be supported (except the combination, where both are None, of course I will raise an Exception then)